src/careamics/config/transformations/transform_union.py

```diff
@@ -0,0 +1,20 @@
+"""Type used to represent all transformations users can create."""
+
+from typing import Union
+
+from pydantic import Discriminator
+from typing_extensions import Annotated
+
+from .n2v_manipulate_model import N2VManipulateModel
+from .xy_flip_model import XYFlipModel
+from .xy_random_rotate90_model import XYRandomRotate90Model
+
+TRANSFORMS_UNION = Annotated[
+    Union[
+        XYFlipModel,
+        XYRandomRotate90Model,
+        N2VManipulateModel,
+    ],
+    Discriminator("name"),  # used to tell the different transform models apart
+]
```

Comment:
This annotation is actually used by Pydantic to select which transform to instantiate (based on the `Discriminator`). 

Now, whether this can be replaced for typing elsewhere in the code, I need to have another look!

Reply:
I see, ok! I need to look at the Pydantic discriminator stuff again, I always forget how it works

Reply:
So indeed for all the uses outside the Pydantic model, we can use the parent class.

But this brought me to then specify exactly what transforms were allowed (since there are only two...), then to write a lot of guarding clauses that became to complex.

So indeed went along with your suggestion to only use a single parameter!